blocks/randompruner.go

```diff
@@ -289,9 +289,12 @@ func (pruner *RandomPruner) prune(ctx context.Context, bytesToPrune uint64) erro
 	return nil
 }
 
-func (pruner *RandomPruner) updatePin(cid cid.Cid) {
+func (pruner *RandomPruner) updatePin(pin cid.Cid) {
+	// clone the Cid to ensure we don't hang on to memory that may come from the network stack
+	// where the Cid was decoded as part of a larger message
+	pin = cid.MustParse(append(make([]byte, 0, len(pin.Bytes())), pin.Bytes()...))
```

Comment:
Could I suggest a clarity change that makes this a little easier to read?

```suggestion
        emptyPinClone := make([]byte, 0, len(pin.Bytes()))
        pinClone := append(emptyPinClone, pin.Bytes()...)
        pin = cid.MustParse(pinClone)
```